validphys2/src/validphys/utils.py

```diff
@@ -5,12 +5,51 @@
 @author: Zahari Kassabov
 """
 import contextlib
+import functools
 import pathlib
 import shutil
 import tempfile
+from typing import Any, Sequence, Mapping
 
 import numpy as np
 from validobj import ValidationError, parse_input
+from frozendict import frozendict
+
+
+# Since typing.Hashable doesn't check recursively you actually
+# have to try hashing it.
+def is_hashable(obj):
+    try:
+        hash(obj)
+        return True
+    except:
+        return False
+
+
+def immute(obj: Any):
```

Comment:
```suggestion
def make_hashable(obj: Any):
```

For symmetry reasongs, since you have `is_hashable`

Reply:
Maybe even `to_hashable` for similarity with `pandas`

Reply:
done in 19771a027d0dacfc966c339180680bdeda06954c